src/elastio-snap.c

```diff
@@ -3569,21 +3569,25 @@ static int snap_read_bio_get_mode(const struct snap_device *dev, struct bio *bio
 static struct inode *elastio_snap_should_mark_bio(struct bio* bio)
 {
 	int i;
-	bio_iter_t iter;
-	bio_iter_bvec_t bvec;
+#ifdef HAVE_BVEC_ITER_ALL
+	struct bvec_iter_all iter;
+	struct bio_vec *bvec;
+	bio_for_each_segment_all(bvec, bio, iter) {
+#else
+	struct bio_vec *bvec;
+	bio_for_each_segment_all(bvec, bio, i) {
+#endif
 
-	bio_for_each_segment(bvec, bio, iter) {
-		for (i = 0; i < inodes_count; i++) {
-			struct inode *inode = page_get_inode(bio_iter_page(bio, iter));
-			if (!inode)
-				continue;
+		struct inode *inode = page_get_inode(bvec->bv_page);
+		if (!inode)
+			continue;
 
+		for (i = 0; i < inodes_count; i++) {
```

Comment:
Lets use another variable, due it can be reused in
`bio_for_each_segment_all(bvec, bio, i) {`